jenkins/bokchoy_db_pull_request.py

```diff
@@ -0,0 +1,287 @@
+    """
+    Checks to see if a file differs from the current version
+    in a repo.
+    Returns true if the files are different.
+    """
+    try:
+        repo_file_contents = repository.get_file_contents(file_path).decoded_content
+    except:
+        raise StandardError(
+            "Could not get contents of file: {}".format(file_path)
+        )
+    return local_file != repo_file_contents
+
+
+def _branch_exists(repository, branch_name):
+    """
+    Checks to see if this branch name already exists
+    """
+    try:
+        repository.get_branch(branch_name)
+    except:
+        return False
+    return True
+
+
+def _create_branch(repository, branch_name, sha):
+    """
+    Create a new branch with the given sha as its head.
+    """
+    try:
+        repository.create_git_ref(branch_name, sha)
+    except:
+        raise StandardError(
+            "Unable to create git branch: {}. "
+            "Check to make sure this branch doesn't already exist.".format(branch_name)
+        )
+
+
+def _update_file(repository, file_path, commit_message, contents, file_sha, branch_name):
+    """
+    Create a commit on a branch that updates the file_path with the string contents.
+    """
+    try:
+        repository.update_file(file_path, commit_message, contents, file_sha, branch_name)
+    except:
+        raise StandardError(
+            "Error updating database file: {}".format(file_path)
+        )
+
+
+def _create_file(repository, file_path, commit_message, contents, branch_name):
+    """
+    Create a commit on a branch that creates a new file with the string contents.
+    """
+    try:
+        repository.create_file(file_path, commit_message, contents, branch_name)
+    except:
+        raise StandardError(
+            "Error creating database file: {}".format(file_path)
+        )
+
+
+def _create_pull_request(repository, title, body, base, head):
+    """
+    Create a new pull request with the changes in head.
+    """
+    try:
+        pull_request = repository.create_pull(
+            title=title,
+            body=body,
+            base=base,
+            head=head
+        )
+    except:
+        raise StandardError(
+            "Could not create pull request"
+        )
+
+
+@click.command()
+@click.option(
+    '--sha',
+    help="Sha of the merge commit to base the new PR off of",
+    required=True,
+)
+@click.option(
+    '--repo_root',
+    help="Path to local edx-platform repository that will "
+         "hold updated database files",
+    required=True,
+)
+def main(sha, repo_root):
+    # Connect to github, and get the edx-platform repo object
+    logger.info("Authenticating with Github")
+    github_token = _get_github_token()
+    github_instance = _authenticate_with_github(github_token)
+    repository = _connect_to_repo(github_instance, "edx-platform")
+
+    # Find the fingerprint thats currently saved on master (if any)
+    fingerprint = _read_fingerprint_from_master(repository)
```

Comment:
This will happen in the first PR (since we haven't check in a fingerprint file yet). That behavior should definitely be fine, and i think the branch name is fine (if thats what youre asking)